app/javascript/gobierto_data/webapp/components/sets/VisualizationsItem.vue

```diff
@@ -0,0 +1,228 @@
+import Button from "./../commons/Button.vue";
+import { getUserId } from "./../../../lib/helpers";
+import { tabs } from '../../../lib/router';
+
+export default {
+  name: "VisualizationsItem",
+  components: {
+    Visualizations,
+    SavingDialog,
+    Button
+  },
+  props: {
+    datasetId: {
+      type: Number,
+      required: true
+    },
+    isUserLogged: {
+      type: Boolean,
+      default: false
+    },
+    privateVisualizations: {
+      type: Array,
+      default: () => []
+    },
+    publicVisualizations: {
+      type: Array,
+      default: () => []
+    },
+    isVizModified: {
+      type: Boolean,
+      default: false
+    },
+    isVizSaved: {
+      type: Boolean,
+      default: false
+    },
+    isVizSavingPromptVisible: {
+      type: Boolean,
+      default: false
+    },
+    enabledVizSavedButton: {
+      type: Boolean,
+      default: false
+    },
+    enabledForkVizButton: {
+      type: Boolean,
+      default: true
+    }
+  },
+  data() {
+    return {
+      labelVisName: I18n.t('gobierto_data.projects.visName') || "",
+      labelEdit: I18n.t('gobierto_data.projects.edit') || "",
+      labelSaveViz: I18n.t('gobierto_data.projects.saveViz') || "",
+      labelVisualize: I18n.t('gobierto_data.projects.visualize') || "",
+      labelDashboard: I18n.t('gobierto_data.projects.dashboards') || "",
+      labelSavedVisualization: I18n.t("gobierto_data.projects.savedVisualization") || "",
+      labelModifiedVizualition: I18n.t("gobierto_data.projects.modifiedVisualization") || "",
+      items: null,
+      config: {},
+      vizID: null,
+      user: null,
+      queryViz: '',
+      isVizElementSavingVisible: false,
+      name: '',
+      isQuerySavingPromptVisible: false,
+      tabs
+    }
+  },
+  watch: {
+    publicVisualizations(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.getDataVisualization(newValue);
+      }
+    },
+    privateVisualization(newValue, oldValue) {
+      if (newValue !== oldValue) {
+        this.getDataVisualization(newValue);
+      }
+    }
+  },
+  async created() {
+    const userId = getUserId()
+    if (userId) {
+      await this.getDataVisualization(this.privateVisualizations);
+      await this.getDataVisualization(this.publicVisualizations);
+    } else {
+      await this.getDataVisualization(this.publicVisualizations);
+    }
+  },
+  beforeDestroy() {
+    //Hide the string, and the buttons return to their initial state.
+    this.$root.$emit("isVizModified", false);
+    this.$root.$emit('disabledSavedVizString')
+  },
+  methods: {
+    onSaveEventHandler(opts) {
+      //Add visualization ID to opts object, we need it to update a viz saved
+      opts.vizID = Number(this.vizID)
+      opts.user = Number(this.user)
+      opts.queryViz = Number(this.queryViz)
+      // get children configuration
+      const config = this.$refs.viewer.getConfig()
+
+      if (!this.isVizSavingPromptVisible) {
```

Comment:
This is the third time (yep, I don't review in order) I see this snippet (you know what that means). But, does it require actually so much complexity just to _save_?